src/plugins/common/src/tools.rs

```diff
@@ -85,6 +85,35 @@ impl From<Vec<Entity>> for Focus {
 	}
 }
 
+pub(crate) fn apply_recursively<'a, TRelatedEntities>(
+	root_entity: Entity,
+	apply_action: &mut impl FnMut(Entity),
+	fetch_related_entities: &'a impl Fn(Entity) -> Option<TRelatedEntities>,
+	is_valid: &impl Fn(Entity) -> bool,
+	should_apply_to_related: &impl Fn(Entity) -> bool,
+) where
```

Comment:
Many arguments. Hard to track, if you have argument order not in head. Maybe create single struct, so we can pass arguments as named arguments to function.

Reply:
done. Although it looks a bit messy when first looking at it. The calls to the function are much clearer now.